openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BotTokenCache.java

```diff
@@ -0,0 +1,76 @@
+package org.openmetadata.service.security.auth;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.CheckForNull;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.entity.teams.AuthenticationMechanism;
+import org.openmetadata.schema.entity.teams.User;
+import org.openmetadata.schema.teams.authn.JWTAuthMechanism;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.jdbi3.UserRepository;
+import org.openmetadata.service.util.EntityUtil;
+import org.openmetadata.service.util.JsonUtils;
+
+@Slf4j
+public class BotTokenCache {
+  private static BotTokenCache INSTANCE;
+  private final LoadingCache<String, String> BOTS_TOKEN_CACHE;
+
+  public BotTokenCache() {
+    BOTS_TOKEN_CACHE =
+        CacheBuilder.newBuilder().maximumSize(1000).expireAfterAccess(2, TimeUnit.MINUTES).build(new BotTokenLoader());
```

Comment:
lets put the expiry to be longer, should be in hours. Ingestion pipelines runs for longer and continue to make requests we want the token to be not fetched from DB every few mins instead lets keep this as an 2hrs